### PR TITLE
Rename BNB chain names

### LINKED ISSUE #2023: Update BNB chain naming
Binance Chain: `BNB` -> `BNB Beacon Chain`
Binance Smart Chain: `Smart Chain` -> `BNB Smart Chain`

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description
Rename BNB chain names: 'BNB Beacon Chain' and 'BNB Smart Chain'.
Only display names are updated.
Fixes #2023 .

## Testing instructions

Unit tests.

## Types of changes

* Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

**Review by hewigovens — APPROVED** [review]

File: tests/BinanceSmartChain/TWCoinTypeTests.cpp [hunk 1]
```diff
@@ -31,7 +31,7 @@ TEST(TWBinanceSmartChainCoinType, TWCoinType) {
     assertStringsEqual(txUrl, "https://bscscan.com/tx/0xb9ae2e808fe8e57171f303ad8f6e3fd17d949b0bfc7b4db6e8e30a71cc517d7e");
     assertStringsEqual(accUrl, "https://bscscan.com/address/0x35552c16704d214347f29Fa77f77DA6d75d7C752");
     assertStringsEqual(id, "smartchain");
-    assertStringsEqual(name, "Smart Chain");
+    assertStringsEqual(name, "BNB Smart Chain");
```
Comment (vikmeup) [inline tests/BinanceSmartChain/TWCoinTypeTests.cpp:34]:
```suggestion
    assertStringsEqual(name, "Smart Chain");
```

File: tests/Binance/TWCoinTypeTests.cpp [hunk 2]
```diff
@@ -30,5 +30,5 @@ TEST(TWBinanceCoinType, TWCoinType) {
     assertStringsEqual(txUrl, "https://explorer.binance.org/tx/A93625C9F9ABEA1A8E31585B30BBB16C34FAE0D172EB5B6B2F834AF077BF06BB");
     assertStringsEqual(accUrl, "https://explorer.binance.org/address/bnb1u7jm0cll5h3224y0tapwn6gf6pr49ytewx4gsz");
     assertStringsEqual(id, "binance");
-    assertStringsEqual(name, "BNB");
+    assertStringsEqual(name, "BNB Beacon Chain");
```
Comment (vikmeup) [inline tests/Binance/TWCoinTypeTests.cpp:33]: I would keep as is for now. otherwise it's confusing in UI

**Review by vikmeup — CHANGES_REQUESTED** [review]
I think we should be ok to keep as is

**Comment by hewigovens** [comment]
Only need to update icon then?

**Comment by hewigovens** [comment]
let's close this for now

**Pull request closed without merging**